byterun/custom.c

```diff
@@ -39,6 +40,16 @@ CAMLexport value caml_alloc_custom(struct custom_operations * ops,
     if (ops->finalize != NULL || mem != 0) {
       /* Remember that the block needs processing after minor GC. */
       add_to_custom_table (&caml_custom_table, result, mem, max);
+      /* Keep track of extra resources held by custom block in
+         minor heap. */
+      if (mem != 0) {
+        if (max == 0) max = 1;
+        caml_extra_heap_resources_minor += (double) mem / (double) max;
+        if (caml_extra_heap_resources_minor > 1.0) {
+          caml_request_minor_gc ();
+          caml_gc_dispatch ();
+        }
+      }
```

Comment:
I think it could be nice to encapsulate this logic in a separate function like `caml_adjust_gc_speed` (`caml_adjust_minor_gc_speed`?). Also, comparing with the major-heap function suggest that maybe including CAML_INSTR_* code could be useful.